### ISSUE TITLE
The function ale#statusline#Count is not working for golang

### ISSUE THREAD

**Issue by YvesZHI** [dense-analysis/ale#3071]
## Information

**VIM version**
vim 8

Operating System: 
Ubuntu 16.04

## What went wrong

I've installed go with the version 1.14.<br>
I've installed three plugins for my vim: YCM, vim-go and ALE.<br>
I executed `:GoInstallBinaries` in my vim to install all of binaries for the plugin vim-go.<br>
I executed `./install.py --go-completer` to compile the YCM and put the ALE-config as below:<br>
```
let g:ale_lsp_root = {'gopls': '~/.vim/bundle/vim-go/bin'}
let g:ale_go_bingo_executable = 'gopls'
let g:ale_completion_enabled = 0
let g:ale_linters = { 'go': ['gofmt', 'golint', 'go vet', 'gopls'] }

let g:go_fmt_command = "goimports"
let g:go_auto_type_info = 1

function! LinterStatus() abort
    let l:counts = ale#statusline#Count(bufnr(''))

    let l:all_errors = l:counts.error + l:counts.style_error
    let l:all_non_errors = l:counts.total - l:all_errors

    return l:counts.total == 0 ? 'OK' : printf(
    \   '%dW %dE',
    \   all_non_errors,
    \   all_errors
    \)
endfunction
set statusline=%{LinterStatus()}
```

However, no matter how I create an error, the status line always shows me an `OK`. The function `ale#statusline#Count` doesn't seem to work as expected.

### :ALEInfo
```
 Current Filetype: go
Available Linters: ['bingo', 'gobuild', 'gofmt', 'golangci-lint', 'golint', 'gometalinter', 'gopls', 'gosimple', 'gotype', 'govet', 'golangserver', 'staticcheck']
   Linter Aliases:
'gobuild' -> ['go build']
'govet' -> ['go vet']
  Enabled Linters: ['gofmt', 'golint', 'gopls', 'govet']
 Suggested Fixers:
  'gofmt' - Fix Go files with go fmt.
  'goimports' - Fix Go files imports with goimports.
  'remove_trailing_lines' - Remove all blank lines at the end of a file.
  'trim_whitespace' - Remove all trailing whitespace characters at the end of every line.
 Linter Variables:

let g:ale_go_go_executable = 'go'
let g:ale_go_golangci_lint_executable = 'golangci-lint'
let g:ale_go_golangci_lint_options = '--enable-all'
let g:ale_go_golangci_lint_package = 0
let g:ale_go_golint_executable = 'golint'
let g:ale_go_golint_options = ''
let g:ale_go_gopls_executable = 'gopls'
let g:ale_go_gopls_options = '--mode stdio'
let g:ale_go_govet_options = ''
let g:ale_go_langserver_executable = 'go-langserver'
let g:ale_go_langserver_options = ''
 Global Variables:

let g:ale_cache_executable_check_failures = v:null
let g:ale_change_sign_column_color = 0
let g:ale_command_wrapper = ''
let g:ale_completion_delay = v:null
let g:ale_completion_enabled = 0
let g:ale_completion_max_suggestions = v:null
let g:ale_echo_cursor = 1
let g:ale_echo_msg_error_str = 'Error'
let g:ale_echo_msg_format = '%code: %%s'
let g:ale_echo_msg_info_str = 'Info'
let g:ale_echo_msg_warning_str = 'Warning'
let g:ale_enabled = 1
let g:ale_fix_on_save = 0
let g:ale_fixers = {}
let g:ale_history_enabled = 1
let g:ale_history_log_output = 1
let g:ale_keep_list_window_open = 0
let g:ale_lint_delay = 200
let g:ale_lint_on_enter = 1
let g:ale_lint_on_filetype_changed = 1
let g:ale_lint_on_insert_leave = 1
let g:ale_lint_on_save = 1
let g:ale_lint_on_text_changed = 'normal'
let g:ale_linter_aliases = {}
let g:ale_linters = {'go': ['gofmt', 'golint', 'go vet', 'gopls']}
let g:ale_linters_explicit = 0
let g:ale_list_vertical = 0
let g:ale_list_window_size = 10
let g:ale_loclist_msg_format = '%code: %%s'
let g:ale_lsp_root = {'gopls': '~/.vim/bundle/vim-go/bin'}
let g:ale_max_buffer_history_size = 20
let g:ale_max_signs = -1
let g:ale_maximum_file_size = v:null
let g:ale_open_list = 0
let g:ale_pattern_options = v:null
let g:ale_pattern_options_enabled = v:null
let g:ale_set_balloons = 0
let g:ale_set_highlights = 1
let g:ale_set_loclist = 1
let g:ale_set_quickfix = 0
let g:ale_set_signs = 1
let g:ale_sign_column_always = 0
let g:ale_sign_error = '>>'
let g:ale_sign_info = '--'
let g:ale_sign_offset = 1000000
let g:ale_sign_style_error = '>>'
let g:ale_sign_style_warning = '--'
let g:ale_sign_warning = '--'
let g:ale_sign_highlight_linenrs = 0
let g:ale_statusline_format = v:null
let g:ale_type_map = {}
let g:ale_use_global_executables = v:null
let g:ale_virtualtext_cursor = 0
let g:ale_warn_about_trailing_blank_lines = 1
let g:ale_warn_about_trailing_whitespace = 1
  Command History:

(executable check - success) gofmt
(finished - exit code 0) ['/bin/bash', '-c', '''gofmt'' -e ''/tmp/vm3Nie2/0/test.go''']

<<<NO OUTPUT RETURNED>>>

(executable check - success) golint
(finished - exit code 0) ['/bin/bash', '-c', '''golint'' ''/tmp/vm3Nie2/1/test.go''']

<<<NO OUTPUT RETURNED>>>

(executable check - success) gopls
(executable check - success) go
(finished - exit code 2) ['/bin/bash', '-c', 'cd ''/home/zyh/tmp/src'' &&  go vet  .']

<<<OUTPUT STARTS>>>
# _/home/zyh/tmp/src
vet: ./test.go:6:2: undeclared name: aaa
<<<OUTPUT ENDS>>>

(started) ['/bin/bash', '-c', '''gopls'' --mode stdio']
```

**Comment by ipkiss42** [comment]
It sounds like the status line is not updated. What if you run `:redrawstatus`?

Maybe you need something like this in your vimrc:

```vim
au User ALELintPost-autocmd rerawstatus
```

**Comment by YvesZHI** [comment]
@ipkiss42  Not working. `:redrawstatus` changed nothing.

**Comment by YvesZHI** [comment]
@ipkiss42   I've succeeded in installing `gopls`, `gofmt` and `golangci-lint` and `:ALEInfo` changed. I've reedited the issue.
```
(executable check - success) gofmt
(finished - exit code 0) ['/bin/bash', '-c', '''gofmt'' -e ''/tmp/vmCvq7C/1/test.go''']

<<<NO OUTPUT RETURNED>>>

(executable check - success) golangci-lint
(finished - exit code 3) ['/bin/bash', '-c', 'cd ''/home/zyh/tmp/src'' && ''golangci-lint'' run ''test.go'' --enable-all']
```

It seems that `golangci-lint` works because it exits with code 3, I think it means that it detected some errors, but still , `ale#statusline#Count` always returns 0.

**Comment by YvesZHI** [comment]
@ipkiss42   I made progress. I've reedited the issue. Please check the new `:ALEInfo`. As you see, I've succeeded in making `gofmt`, `golint`, `gopls` work, but I still can not get any hint of error.

**Comment by w0rp** [comment]
`ale#statusline#Count` itself works just fine. I don't support any kind of statusline support directly, as it's far too complicated and would take too much time. I recommend using [lightline](https://github.com/itchyny/lightline.vim) with [lightline-ale](https://github.com/maximbaz/lightline-ale). I haven't tried recent commits of lightline myself, though. I have used [an old fork I edited](https://github.com/w0rp/lightline.vim) for 3 years now. I seldom update Vim plugins if they continue to work.
